### PR TITLE
C24_WMDE_Desktop_EN_02b (round2)

### PR DESCRIPTION
**Pull request by moiikana**
round 2 of the EN02 banner

- [x] I already donated 
- [x] soft close
- [x] progress bar
---
- [x] tests
- [X] alreadydonatedmodal code is still used by some precampaign banners and thus can't be removed until January 2025
- [x] contains fixes for payment icons and UoF graphics
---
- round 1 of this banner is here: https://github.com/wmde/fundraising-banners/pull/572

https://phabricator.wikimedia.org/T375601

### CODE REVIEW

**Comment by moiikana** [comment]
still needs to be rebased properly

**Comment by moiikana** [comment]
contains the latest UoF changes from https://github.com/wmde/fundraising-banners/pull/563 :heavy_check_mark: 
- [x] banner ctrl
- [x] banner var
- [x] fallback banner

**Comment by Sperling-0** [comment]
⚠️ Rebase this PR from latest main.

To get the latest UoF working:

- For Desktop banners: `BannerCtrl.vue`, `BannerVar.vue`, `FallbackBanner.vue` needs to be updated. (If it is DE desktop include DE infographic and vice versa)
- For Mobile/ Ipad banners: `BannerCtrl.vue`, `BannerVar.vue` needs to be updated. (If it is DE mobile include DE infographic and vice versa)

**Comment by moiikana** [comment]
rebased main,
contains the latest BIC changes https://phabricator.wikimedia.org/T377589

**Comment by moiikana** [comment]
rebased main

**Review by Abban — APPROVED** [review]
Looks good, I just spotted a few things with the progress bar.

You should update the variable for the progress bar in the dark colour swatch to:

```--progress-bar-wrapper-background: #{$grey700};```

In the theme's ProgressBarAlternative.scss on line 69 you should increase the line height to 26px. It'll fix the alignment in the left text:

Screenshot 2024-10-24 at 14 51 29

File: banners/english/C24_WMDE_Desktop_EN_02b/styles/styles.scss [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+@use 'src/themes/Treedip/DonationForm/MultiStepDonation';
+@use 'src/themes/Treedip/DonationForm/BubbleForm/SelectGroupBubble';
+@use 'src/themes/Treedip/DonationForm/BubbleForm/SelectGroupRadioBubbles' with (
+	$height: 34px
+);
+@use 'src/themes/Treedip/DonationForm/BubbleForm/SelectCustomAmountRadioBubble' with (
+	$height: 34px
+);
+@use 'src/themes/Treedip/DonationForm/BubbleForm/SelectGroupBubbleImageLabel';
+@use 'src/themes/Treedip/DonationForm/BubbleForm/SelectGroupBubbleImageLabel_dark';
```
Comment (Abban) [inline banners/english/C24_WMDE_Desktop_EN_02b/styles/styles.scss]: You can delete this import, the swatching improvements handle the payment icon colours differently now. If you want you can also delete the file and the imports from the older EN desktop banners

**Comment by moiikana** [comment]
rebased because https://github.com/wmde/fundraising-banners/pull/597 was merged

**Comment by moiikana** [comment]
rebased main

**Comment by moiikana** [comment]
another main rebase

**Review by gbirke — APPROVED** [review]
I like the single-commit-for-single-feature-activation of this PR. It'll help later when looking at the history of a banner. To avoid making the titles look like we activated a feature for all campaigns we might want to prefix the title with the campaign name next time. If you like you could rebase-and-edit the existing commit titles, but we can also leave them as-is

**Pull request merged**